docs/development/creating-editing-renovate-presets.md

```diff
@@ -0,0 +1,46 @@
+# Creating/editing Renovate presets
+
+Renovate comes with default presets that you can find in the `lib/config/presets/internal` directory.
+You can suggest changes to the presets with a pull request.
+
+Follow the rules below to increase the chance that your pull request gets merged.
+
+## General rules
+
+1. Only create/edit presets for problems which can not be fixed upstream
+1. The internal preset should be helpful for a significant number of Renovate users
+
+### Specific rules
+
+#### Group presets
+
+We have multiple kinds of `group:` presets, with different rules.
+
+##### Rules for `group:monorepo` preset
+
+1. Only group dependencies that _must_ be updated together
+
+##### Rules for `group:recommended` presets
+
+1. The `group:recommended` preset is for related dependencies which aren't from a monorepo but which usually need to be updated together as separate PRs may each break
+
+##### Rules for `group:*` presets
+
+1. Finally, any other `group:*` presets can be added if they are beneficial to a wide number of users
+1. They don't need to be added to `group:recommended`, meaning that users will "opt in" to them one by one and not get them automatically from `config:base`, which includes `group:monorepo` and `group:recommended`
+
+#### Replacement presets
+
+Rules:
+
+1. Replacement PRs should ideally propose a replacement only once the user is on a compatible version, by specifying a compatible `matchCurrentVersion` constraint
```

Comment:
Maybe we should mention to limit the range if there are multiple `replacements` / package renames? We needed to "narrow the version range" for the `joi` to `@hapi/joi` replacement, so that people didn't get downgraded to a version of `joi` they upgraded past already...

See PR https://github.com/renovatebot/renovate/pull/13205

Reply:
I think that one's probably a rarely one where there were multiple major versions each with both packages?

Reply:
That whole `joi`/`@hapi/joi` and then back to `joi` thing seems to be an edgecase.

Maybe we can leave out the `, by specifiying a compatible matchCurrentVersion` step in the instructions?